samtranslator/model/apigatewayv2.py

```diff
@@ -73,6 +73,7 @@ def __init__(
         identity=None,
         authorizer_payload_format_version=None,
         enable_simple_responses=None,
+        is_aws_iam_authorizer=False,
```

Comment:
As an alternative to this we could also define a new `type` field which could be one of the following supported values:

1. `None` - If not supplied the authorizer type will be autodetected like it is now
2. `AWS_IAM`
3. `JWT`
4. `LAMBDA`

This would then look like:

```
  ServerlessHttpApi:
    Type: AWS::Serverless::HttpApi
    Properties:
      Auth:
        Authorizers:
          IamAuthorizer:
            Type: "AWS_IAM"
          JwtAuthorizer:
            Type: "JWT"
          LambdaAuthorizer:
            Type: "LAMBDA"
          AutoDetectJwt:
            ... jwt properties ...
          AutoDetectLambda:
            ... lambda properties ...
```

Reply:
Note: I played around with this locally and didn't like it. Decided on #1878 instead as an alternate approach.